tests/apollo/test_skvbc_view_change.py

```diff
@@ -600,6 +600,71 @@ async def test_recovering_of_replicas_with_initiated_view_change(self, bft_netwo
         await bft_network.wait_for_fast_path_to_be_prevalent(
             run_ops=lambda: skvbc.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: c == 0, rotate_keys=True)
+    @verify_linearizability()
+    async def test_recovering_of_replicas_with_failed_quorum_view_change(self, bft_network, tracker):
+        """
+        We need to verify the system correctly handles the case where we have F+1 replicas unavailable (stopped).
+         In this scenario the system will temporarily lose liveness, but we need to verify that once the stopped replicas are up again the system will recover and process client requests again.
+
+        1. Starts all replicas.
+        2. Send Client requests to verify system is working correctly.
+        3. Stop F+1 Replicas excluding the primary.
+        4. Send additional client requests that won’t be processed due to lack of quorum,
+         but will trigger the live replicas to move to a higher view.
+        5. Verify the 2F remaining replicas move to the next view.
+         Note - we cannot use bft_network.wait_for_view since it waits for a quorum (2F+1)
+          of replicas to enter the view and we will only have 2F.
+           We can use something similar to what we have in Get Replica View.
+            There is also the more generic getter that was added later metric getter.
+        6. Start the previously stopped replicas.
+        7. Verify the system recovers and all replicas enter the new view (here we can use bft_network.wait_for_view).
+        """
+
+        # start replicas
+        [bft_network.start_replica(i) for i in bft_network.all_replicas()]
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        # write to trigger vc
+        await skvbc.run_concurrent_ops(10)
+
+        initial_primary = await bft_network.get_current_primary()
+        await bft_network.wait_for_replicas_to_reach_view(bft_network.all_replicas(), 1)
+
+        to_stop = random.sample(
+            bft_network.all_replicas(without={initial_primary}), bft_network.config.f + 1)
+
+        # stop replicas
+        [bft_network.stop_replica(i) for i in to_stop]
+
+        # write to trigger vc
+        await skvbc.run_concurrent_ops(10)
+
+        running_replicas = []
+        for i in bft_network.all_replicas():
+            is_in_to_stop = False
+            for j in to_stop:
+                if (i == j):
+                    is_in_to_stop = True
+                    break
+            if (not is_in_to_stop):
+                running_replicas.append(i)
+
+        for r in running_replicas:
+            view_of_replica = 0
+            while view_of_replica == 0:
+                view_of_replica = await self._get_gauge(r, bft_network, 'view')
+                await trio.sleep(seconds=0.1)
+            self.assertEqual(view_of_replica, 2, "Replica failed to reach expected view")
```

Comment:
I think something like `await wait_for_replicas_to_reach_view(bft_network.all_replicas(without={to_stop}), 2)` should be able to achieve the same results. Or you could simply iterate over all replicas and continue if the current one is present in `to_stop`.